src/solver/variable/economy/STStorageCashFlowByCluster.h

```diff
@@ -0,0 +1,305 @@
+/*
+** Copyright 2007-2023 RTE
+** Authors: Antares_Simulator Team
+**
+** This file is part of Antares_Simulator.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** There are special exceptions to the terms and conditions of the
+** license as they are applied to this software. View the full text of
+** the exceptions in file COPYING.txt in the directory of this software
+** distribution
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+#pragma once
+
+#include "../variable.h"
+
+namespace Antares::Solver::Variable::Economy
+{
+struct VCardSTstorageCashFlowByCluster
+{
+    //! Caption
+    static std::string Caption()
+    {
+        return "STS Cashflow";
+    }
+    //! Unit
+    static std::string Unit()
+    {
+        return "CashFlow - Euro";
+    }
+
+    //! The short description of the variable
+    static std::string Description()
+    {
+        return "Cash Flow for short term storage";
+    }
+
+    //! The expecte results
+    typedef Results<R::AllYears::Average< // The average values throughout all years
+      >>
+      ResultsType;
+
+    //! The VCard to look for for calculating spatial aggregates
+    typedef VCardSTstorageCashFlowByCluster VCardForSpatialAggregate;
+
+    enum
```

Comment:
can we use class enum to avoid misusage e.g enum conversion to int?

Reply:
No
```
/home/omnesflo/Antares_Simulator/src/libs/antares/study/../../../solver/variable/economy/STStorageCashFlowByCluster.h:61:5: error: unnamed scoped enum is not allowed
   61 |     {
      |     ^
compilation terminated due to -Wfatal-errors.
```

We'd have to name this enum (e.g. ScopedEnum), but then the caller would need to call `VCardSTstorageCashFlowByCluster::ScopedEnum::categoryDataLevel` instead of just `VCardSTstorageCashFlowByCluster::categoryDataLevel` right now.

I think it's too much of a hassle to change this for all output variables.